resource-managers/kubernetes/core/src/test/scala/org/apache/spark/shuffle/KubernetesLocalDiskShuffleDataIOSuite.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle
+
+import scala.concurrent.duration._
+
+import org.scalatest.concurrent.Eventually.{eventually, interval, timeout}
+
+import org.apache.spark.{LocalSparkContext, MapOutputTrackerMaster, SparkConf, SparkContext, SparkFunSuite, TestUtils}
+import org.apache.spark.LocalSparkContext.withSpark
+import org.apache.spark.deploy.k8s.Config.KUBERNETES_DRIVER_REUSE_PVC
+import org.apache.spark.internal.config._
+import org.apache.spark.scheduler.cluster.StandaloneSchedulerBackend
+
+class KubernetesLocalDiskShuffleDataIOSuite extends SparkFunSuite with LocalSparkContext {
+
+  val conf = new SparkConf()
+    .setAppName("ShuffleExecutorComponentsSuite")
+    .setMaster("local-cluster[1,1,1024]")
+    .set(UI.UI_ENABLED, false)
+    .set(DYN_ALLOCATION_ENABLED, true)
+    .set(DYN_ALLOCATION_SHUFFLE_TRACKING_ENABLED, true)
+    .set(DYN_ALLOCATION_INITIAL_EXECUTORS, 1)
+    .set(DYN_ALLOCATION_MIN_EXECUTORS, 1)
+    .set(IO_ENCRYPTION_ENABLED, false)
+    .set(KUBERNETES_DRIVER_REUSE_PVC, true)
+    .set(SHUFFLE_IO_PLUGIN_CLASS, classOf[KubernetesLocalDiskShuffleDataIO].getName)
+
+  test("recompute is not blocked by the recovery") {
+    sc = new SparkContext(conf)
+    withSpark(sc) { sc =>
+      val master = sc.env.mapOutputTracker.asInstanceOf[MapOutputTrackerMaster]
+      assert(master.shuffleStatuses.isEmpty)
+
+      val rdd = sc.parallelize(Seq((1, "one"), (2, "two"), (3, "three")), 3)
+        .groupByKey()
+      rdd.collect()
+      assert(master.shuffleStatuses.keys.toSet == Set(0))
+      val loc1 = master.shuffleStatuses(0).mapStatuses(0).location
```

Comment:
loc1 is not used?

Reply:
Yes, `loc1` is not used in this test case. I'll remove it.